### PR TITLE
Refactoring of python-can CANSockets [ready for merge]

### PR DESCRIPTION
**Pull request by polybassa**
This PR requires #2428 to be merged.

Complete refactoring of the usage of `python-can` inside Scapy.

In the past, `NativeCANSockets` and `PythonCANSockets` showed a different runtime behaviour. 
This PR aims to imitate the behaviour of Linux SocketCAN for `BusABC` objects of `python-can`. This includes the functionality of `select` which is no possible on `python-can` `Bus` objects.
Additionally to that, in the past it was not possible to create multiple `CANSocket()` objects with the same `BusABC` object as interface. `BusABC` implements a abstraction for all kind of drivers for CAN-interfaces. Some of these drives can only be used once. The new middle layer introduced in this PR resolves this problem. It's now possible to create as many `PythonCANSockets()` as you want, identically to the behaviour of `NativeCANSockets`. 

This PR removes `srcan`.

### CODE REVIEW

**Comment by polybassa** [comment]
This PR requires #2435 and #2456

File: scapy/contrib/cansocket_python_can.py [hunk 1]
```diff
@@ -55,38 +163,27 @@ def recv_raw(self, x=0xffff):
...
-            return self.iface.send(msg)
-        except can_Error as ex:
-            raise ex
+        msg = can_Message(is_remote_frame=x.flags == 0x2,
+                          is_extended_id=x.flags == 0x4,
+                          is_error_frame=x.flags == 0x1,
+                          arbitration_id=x.identifier,
+                          dlc=x.length,
+                          data=bytes(x)[8:])
+        if hasattr(x, "sent_time"):
```
[the remark this answers is not in the document]
Comment (polybassa) [reply scapy/contrib/cansocket_python_can.py]: Thanks for the hint

File: scapy/contrib/cansocket_python_can.py [hunk 2]
```diff
@@ -12,38 +12,146 @@
...
 from can import Message as can_Message
-from can import CanError as can_Error
+from can import CanError as can_CanError
+from can import BusABC as can_BusABC
+from can.interface import Bus as can_Bus
+
+if six.PY2:
+    import Queue as queue
+else:
+    import queue as queue
```
[the remark this answers is not in the document]
Comment (polybassa) [reply scapy/contrib/cansocket_python_can.py]: Thanks

**Comment by polybassa** [comment]
Is this related to my changes? https://travis-ci.com/secdev/scapy/jobs/285359096#L1114

**Comment by polybassa** [comment]
I've no glue, why this test fails only on Windows: https://ci.appveyor.com/project/secdev/scapy/builds/30692206/job/bmt58d1ugk1o9pvi#L2707
@gpotter2 do you have an idea?

File: scapy/contrib/cansocket.py [hunk 3]
```diff
@@ -29,18 +29,20 @@
     log_loading.info("Configuration 'conf.contribs['CANSocket'] not found.")
 
 
+from scapy.contrib.cansocket_native import (CAN_FRAME_SIZE, CAN_INV_FILTER)  # noqa: E501 F401
```
Comment (gpotter2) [inline scapy/contrib/cansocket.py]: watch out your imports. This will import `scapy.arch.linux`, even on windows

Comment (polybassa) [reply scapy/contrib/cansocket.py]: Sounds reasonable. Thanks for your time. I'll test it

Comment (polybassa) [reply scapy/contrib/cansocket.py]: Fixed... Thanks a lot

Comment (gpotter2) [reply scapy/contrib/cansocket.py]: 👍

**Comment by polybassa** [comment]
@gpotter2 How can I fix codacy? Is this fix-able or can you use your admin power to fix it?

**Comment by polybassa** [comment]
@gpotter2 could you please have a look at this tests: https://travis-ci.com/secdev/scapy/jobs/287828075#L4300
It's failing because of something related to timestamps.

**Comment by polybassa** [comment]
found the answer for the failing tests in #2482

**Pull request merged**